File: code_samples/elixir/design_patterns_factory.ex
# Design Patterns: Factory
# AI/ML Training Sample

defmodule Factory do
  defstruct data: ""
  
  def new(), do: %Factory{}
  
  def process(%Factory{} = struct, input) do
    %{struct | data: input}
  end
  
  def get_data(%Factory{data: data}), do: data
  
  def validate(%Factory{data: data}) do
    String.length(data) > 0
  end
end

# Example usage
instance = Factory.new()
updated = Factory.process(instance, "example")
IO.puts("Data: " <> Factory.get_data(updated))
IO.puts("Valid: " <> to_string(Factory.validate(updated)))

File: code_samples/typescript/design_patterns_factory.ts
/**
 * Design Patterns: Factory
 * AI/ML Training Sample
 */

interface IFactory {
    data: string;
    process(input: string): void;
    validate(): boolean;
}

class Factory implements IFactory {
    data: string;
    
    constructor() {
        this.data = "";
    }
    
    process(input: string): void {
        this.data = input;
    }
    
    getData(): string {
        return this.data;
    }
    
    validate(): boolean {
        return this.data.length > 0;
    }
}

// Example usage
const instance = new Factory();
instance.process("example");
console.log(`Data: ${instance.getData()}`);
console.log(`Valid: ${instance.validate()}`);

export { Factory, IFactory };
